pkg/crawler/crawler.go

```diff
@@ -5,31 +5,47 @@ import (
 	"encoding/hex"
 	"encoding/xml"
 	"fmt"
-	"github.com/aquasecurity/trivy-java-db/pkg/fileutil"
-	"github.com/aquasecurity/trivy-java-db/pkg/types"
 	"io"
 	"log"
 	"net/http"
+	"os"
 	"path/filepath"
+	"sort"
 	"strings"
 	"sync"
+	"time"
+
+	"github.com/aquasecurity/trivy-java-db/pkg/fileutil"
+	"github.com/aquasecurity/trivy-java-db/pkg/types"
+	"github.com/google/licenseclassifier/v2/tools/identify_license/backend"
+	"github.com/samber/lo"
 
 	"github.com/PuerkitoBio/goquery"
 	"github.com/hashicorp/go-retryablehttp"
+	cmap "github.com/orcaman/concurrent-map/v2"
+	"golang.org/x/net/html/charset"
 	"golang.org/x/sync/semaphore"
 	"golang.org/x/xerrors"
 )
 
 const mavenRepoURL = "https://repo.maven.apache.org/maven2/"
 
 type Crawler struct {
-	dir  string
-	http *retryablehttp.Client
+	dir        string
+	licensedir string
+	http       *retryablehttp.Client
 
 	rootUrl string
 	wg      sync.WaitGroup
 	urlCh   chan string
 	limit   *semaphore.Weighted
+	opt     Option
+
+	// license classifier
+	classifier *backend.ClassifierBackend
+
+	// uniqueLicenseKeys : key is hash of license url or name in POM, whichever available
```

Comment:
Why do we need hash here?

Reply:
We saw scenarios where individuals have dumped full license files and invalid characters in the license name/url field within pom.xml so a  hash ensures the key is consistent and short.

We dump the same hash in the indexes cache file and use it during the build stage to map/update the license information from the license cache files which track the license key hash to classified license string mapping.

Reply:
I believe this approach allows us to avoid using digests.
https://github.com/aquasecurity/trivy-java-db/pull/21#discussion_r1329645135